### PR TITLE
Add ISSUER variable to central OPA deployment

### PR DESCRIPTION
**Pull request by tpoliaw**
Only remembered we'd added this to the token validation after the other PR was merged. Not sure what the issuer  value should be so I opted for the value used in the docs/examples.

### CODE REVIEW

**Comment by garryod** [comment]
We should probably use the OIDC discovery endpoint to get these, see: https://openid.net/specs/openid-connect-discovery-1_0.html

**Comment by tpoliaw** [comment]
From the OPA docs [here](https://www.openpolicyagent.org/docs/latest/oauth-oidc/#metadata-discovery) it looks like we should be setting ISSUER but not JWKS_ENDPOINT

File: charts/opa/Chart.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ apiVersion: v2
 name: opa
 description: An OPA deployment to run alongside applications requiring authorization
 type: application
-version: 0.5.0
+version: 0.5.1
```
Comment (garryod) [inline charts/opa/Chart.yaml:5]: This should be a breaking change (`0.6.0`)

File: policy/diamond/policy/token/token.rego [hunk 2]
```diff
@@ -2,17 +2,25 @@ package diamond.policy.token
 
 import rego.v1
 
+issuer := opa.runtime().env.ISSUER
+
+jwks_endpoint := jwks_endpoint if {
+	metadata := http.send({
+		"url": concat("", [issuer, "/.well-known/openid--configuration"]),
```
Comment (garryod) [inline policy/diamond/policy/token/token.rego:9]: We should pass in the well known URL and retrieve the `issuer` from there, as the path of the well-known endpoint isn't guaranteed to be `/.well-known/openid-configuration`

Comment (garryod) [inline policy/diamond/policy/token/token.rego:9]: Not sure the double hyphen is correct

**Review by garryod — CHANGES_REQUESTED** [review]

File: policy/diamond/policy/token/token.rego (hunk 2 above)
Comment (garryod) [reply policy/diamond/policy/token/token.rego]: Oh, fair enough then. Swear I've seen it be something else but can't figure out where

**Review by garryod — DISMISSED** [review]

**Review by garryod — APPROVED** [review]

**Comment by garryod** [comment]
> Sorry, found a docs change I missed in the final read through

No worries, much preferred to finding it six months from now

**Pull request merged**